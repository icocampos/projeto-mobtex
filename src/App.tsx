import { useGet } from './hooks/useGet';

export function App() {
  const { data, loading, error } = useGet('teste')
    if (loading) return <p>Carregando</p>
    if (error) return <p>Ocorreu um erro</p>
  console.log(data);
  return(
    <ul>
      {data && data.data.map(item => <li key={item.id} >{item.name}</li>)}
    </ul>

  );
};

if (process.env.NODE_ENV === 'development') {
  const { worker } = require('./mocks/browser')
  worker.start()
}
