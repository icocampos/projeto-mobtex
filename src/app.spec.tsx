import { render, screen, waitFor } from "@testing-library/react";
import { App } from "./App";
import  renderer from 'react-test-renderer';

describe('App', () => {
  it('deve renderizar o componente', () => {
    const tree = renderer
    .create(<App/>)
    .toJSON()
    expect(tree).toMatchSnapshot()
    render(<App />)
  })

  it('deve renderizar a lista', async () => {
    render(<App />)
    await waitFor (() => screen.getByText('Olimpiadas App'))
  })
})

