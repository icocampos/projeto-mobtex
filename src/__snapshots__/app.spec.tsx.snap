// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App deve renderizar o componente 1`] = `
<h1
  className="sc-bdfBQB eNUYty"
>
  Olimpiadas App
</h1>
`;
